[2024-04-26T08:06:36.299+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-26T08:06:36.309+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stream-dag.get_objects manual__2024-04-26T08:06:34.865345+00:00 [queued]>
[2024-04-26T08:06:36.316+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stream-dag.get_objects manual__2024-04-26T08:06:34.865345+00:00 [queued]>
[2024-04-26T08:06:36.316+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 1
[2024-04-26T08:06:36.325+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): get_objects> on 2024-04-26 08:06:34.865345+00:00
[2024-04-26T08:06:36.332+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=1335) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-04-26T08:06:36.334+0000] {standard_task_runner.py:63} INFO - Started process 1337 to run task
[2024-04-26T08:06:36.334+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'stream-dag', 'get_objects', 'manual__2024-04-26T08:06:34.865345+00:00', '--job-id', '9', '--raw', '--subdir', 'DAGS_FOLDER/new-dag.py', '--cfg-path', '/tmp/tmptaka7ou4']
[2024-04-26T08:06:36.336+0000] {standard_task_runner.py:91} INFO - Job 9: Subtask get_objects
[2024-04-26T08:06:36.370+0000] {task_command.py:426} INFO - Running <TaskInstance: stream-dag.get_objects manual__2024-04-26T08:06:34.865345+00:00 [running]> on host 701d3b82b873
[2024-04-26T08:06:36.432+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='stream-dag' AIRFLOW_CTX_TASK_ID='get_objects' AIRFLOW_CTX_EXECUTION_DATE='2024-04-26T08:06:34.865345+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-04-26T08:06:34.865345+00:00'
[2024-04-26T08:06:36.434+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-26T08:06:37.077+0000] {python.py:237} INFO - Done. Returned value was: {'ResponseMetadata': {'RequestId': 'VEXS5XFDP7YCV1YQ', 'HostId': 'zBMCsF91K22EzSx8VZVuunQiKSibPwCJ/jlEMMGtQDKfZmkHcPDeQ8TltUt3e3EVXf8TR7LIate/LyOX24SsQw==', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amz-id-2': 'zBMCsF91K22EzSx8VZVuunQiKSibPwCJ/jlEMMGtQDKfZmkHcPDeQ8TltUt3e3EVXf8TR7LIate/LyOX24SsQw==', 'x-amz-request-id': 'VEXS5XFDP7YCV1YQ', 'date': 'Fri, 26 Apr 2024 08:06:37 GMT', 'last-modified': 'Fri, 26 Apr 2024 07:13:25 GMT', 'etag': '"c3f6e6767a50ce00803dcd08326ac234"', 'x-amz-server-side-encryption': 'AES256', 'accept-ranges': 'bytes', 'content-type': 'binary/octet-stream', 'server': 'AmazonS3', 'content-length': '151'}, 'RetryAttempts': 0}, 'AcceptRanges': 'bytes', 'LastModified': datetime.datetime(2024, 4, 26, 7, 13, 25, tzinfo=tzutc()), 'ContentLength': 151, 'ETag': '"c3f6e6767a50ce00803dcd08326ac234"', 'ContentType': 'binary/octet-stream', 'ServerSideEncryption': 'AES256', 'Metadata': {}, 'Body': <botocore.response.StreamingBody object at 0x7f1fcd709690>}
[2024-04-26T08:06:37.077+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-26T08:06:37.083+0000] {xcom.py:677} ERROR - Object of type StreamingBody is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2024-04-26T08:06:37.084+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/serialization/serde.py", line 189, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'botocore.response.StreamingBody'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 486, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 247, in set
    value = cls.serialize_value(
            ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 675, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type StreamingBody is not JSON serializable
[2024-04-26T08:06:37.091+0000] {taskinstance.py:1205} INFO - Marking task as FAILED. dag_id=stream-dag, task_id=get_objects, execution_date=20240426T080634, start_date=20240426T080636, end_date=20240426T080637
[2024-04-26T08:06:37.101+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 9 for task get_objects (Object of type StreamingBody is not JSON serializable; 1337)
[2024-04-26T08:06:37.109+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-04-26T08:06:37.121+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-26T08:06:37.122+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
